### DISCUSSION TITLE
Add/remove NES headers & convert .md files to .bin format

### DISCUSSION THREAD

**Discussion by KeyRogue** [btimofeev/UniPatcher#76] [Ideas]
Is it possible for these two to be implemented in the future? Since these are helpful for those who have no access to PC.

**Comment by Europia79** [comment]
For **Sega Genesis / Mega Drive,** the extensions `.68k .bin .gen .md` all have the same file format & are interchangeable.
However, the extension `.smd` (which stands for **Super Magic Drive**) is an interleaved format.

As far as NES patches go, that's one of the major problems:  Is that often times, romhacking authors will only create a patch against whatever ROM they happen to have on hand:  And the Source ROM and your ROM only differ in which header they have: (1) iNes 1.0 (2) iNes 2.0 (3) signed iNes (4) other/bad iNes (5) or even no header at all.

So that would be a pretty useful feature:  It's actually the exact reason why I asked for a CLI (Command Line Interface) feature:
...To save/remove/restore NES headers via script.

The magic numbers for iNes headers are `4E45531A` and `4644531A`

**Comment by btimofeev** [comment]
Yes, this can be implemented in the future.

**Comment by Europia79** [comment]
Here's the information for `.smd` **Super Magic Drive** interleaved format:
https://segaretro.org/SMD#:~:text=SMD%20is%20an%20interleaved%20file,why%20the%20data%20becomes%20interleaved.

Looks like it reads the ROM in blocks of 16K with the 1st 8K being the even bytes & the last 8K being the odd bytes.

Bidirectional conversion would actually be pretty useful because some romhacks & translations actual require this rare format.

So, you would actually have to convert from `.bin` to `.smd`, apply the patch, then convert back to `.bin`

(where `.bin` is actually synonymous with _ANY_ of the Sega Genesis/MD extensions: `.68k`, `.bin`, `.gen`, `.md`)

**Comment by Europia79** [comment]
Sorry, after taking a cursory examination of one such example that requires .smd format, the Sega Retro information appears to be incorrect (or incomplete):  https://www.romhacking.net/hacks/4038/

Just eyeballing it, `.smd` appears to have a 0x200 (512) byte header with the 1st 16-bytes as `40 03 00 00 00 00 00 00 AA BB 06 00 00 00 00 00` ???  ...I'll have to look into it more & research the spec.
